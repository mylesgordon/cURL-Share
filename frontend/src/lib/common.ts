export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}