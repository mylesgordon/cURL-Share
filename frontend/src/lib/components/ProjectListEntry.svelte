<script lang="ts">
	import { Card } from 'agnostic-svelte';

	export let description: string;
	export let id: number;
	export let name: string;
	const dataTestIdPrefix = `project-list-entry-${id}`;
	const link = `/project/${id}`;
</script>

<Card isBorder isRounded isShadow css="mt-4">
	<a href={link} class="p-4 w-48 truncate" data-testid={dataTestIdPrefix + '-title'}>
		{name}
	</a>
	<p class="p-4 flex-grow flex-shrink" data-testid={dataTestIdPrefix + '-description'}>
		{description}
	</p>
</Card>
