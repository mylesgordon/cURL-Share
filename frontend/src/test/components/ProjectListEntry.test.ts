import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/svelte';
import { screen } from '@testing-library/dom';
import { testMatchingSnapshot } from '../common';
import ProjectListEntry from '$lib/components/ProjectListEntry.svelte';
import type { ComponentProps } from 'svelte';

describe('ProjectListEntry component', () => {
	const props: ComponentProps<ProjectListEntry> = {
		description: 'Test description',
		id: 1,
		name: 'Project title',
		index: 0,
		length: 1
	};
	testMatchingSnapshot(ProjectListEntry, props);

	it('should render title and description correctly', () => {
		render(ProjectListEntry, props);
		const description = screen.getByTestId('project-list-entry-1-description');
		const title = screen.getByTestId('project-list-entry-1-title');
		expect(description).toHaveTextContent('Test description');
		expect(title).toHaveTextContent('Project title');
	});

	it('should link to the correct place', () => {
		render(ProjectListEntry, props);
		const title = screen.getByTestId('project-list-entry-1-title');
		expect(title.getAttribute('href')).toBe('/project/1');
	});
});
