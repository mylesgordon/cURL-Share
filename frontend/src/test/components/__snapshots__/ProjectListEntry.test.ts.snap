// Vitest Snapshot v1

exports[`ProjectListEntry component > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="card card-shadow card-rounded card-border mt-4 s-CTyDWs24qi3Z"
    >
      <a
        class="p-4 w-48 truncate"
        data-testid="project-list-entry-1-title"
        href="/project/1"
      >
        Project title
      </a>
       
      <p
        class="p-4 flex-grow flex-shrink"
        data-testid="project-list-entry-1-description"
      >
        Test description
      </p>
    </div>
    <!--&lt;Card&gt;-->
    <!--&lt;ProjectListEntry&gt;-->
  </div>
</body>
`;
